### DISCUSSION TITLE
P_hf in HeatTransferFromHeatStructure1Phase

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#25492] [Q&A General]
Hello. @joshuahansel @GiudGiud 
I want to know what the `P_hf`'s effect is. If I define a `HeatStructureCylindrical` connected with two `FlowChannel1Phase`. And I define two `HeatTransferFromHeatStructure1Phase`. When I set the P_hf to 1 in the first HeatTransfer and set the P_hf to 2 in the second. Will the power of the `HeatStructureCylindrical` be divided into two parts(1/3 and 2/3) for the two `FlowChannel1Phase`?
I've tried to run a case but it seems the P_hf can't achieve the above things.

Ethan

**Comment by GiudGiud** [comment]
Hello

From the docs
```
The parameter ["P_hf"](https://mooseframework.inl.gov/source/components/HeatTransferFromHeatStructure1Phase.html#acf4cc4e-ab98-4ab4-8d55-6d003fbee4ff) is optional and specifies the heated perimeter 

if unspecified, this is computed from the cross-sectional area assuming a circular cross section.
```

so basically the perimeter * the heat transfer coeff * the temperate delta gets you the heat flux.
I think you cant just give an arbitrary value there, you need to get P_hf to be the size it should be for the flux you are trying to get

Guillaume

**Reply by Ethan-xj** [reply]
It seems quite different than I thought. So if I want to build a model: a heat structure connected with two flow channels. How should I set the fraction of the power?

**Reply by GiudGiud** [reply]
for the fraction of heat flux you mentioned you can set Phf1 = 2 * Phf2 with : 
- the same heat transfer coefficient for both
- the same temperature profiles in both

but the base value of Phf1 / the heat transfer coefficients have to be set to get the heat flux you want. Usually it s just:
- Phf1 is the real heated perimeter
- Hw is the heat transfer coefficient from a correlation / closure

Can you attach a heat structure to 2 channels btw? I've never done that

**Reply by Ethan-xj** [reply]
Yes, I can. I set two `HeatTransferFromHeatStructure1Phase` whose `hs` is the same heat structure and `flow_channel` is different. But when I set like that, I think the Phf is not working right. When I set the total power and the heat structure, the `volumetric heat source` is defined. And the Phf*`volumetric heat source` should be the linear power density using in flow channel.
But I set the model: one hs and one flow channel. And I change the Phf, the rise of the temperature is not changing.

**Reply by GiudGiud** [reply]
if the Phf is already large enough for all the heat to make its way into the channel, then raising it more will not change much

**Reply by joshuahansel** [reply]
> But I set the model: one hs and one flow channel. And I change the Phf, the rise of the temperature is not changing.

So to be clear, you have issues with just a single HS/flow channel? The rate of heat transfer is proportional to it, with a given HTC and temperature difference. If you're seeing that the temperature distribution over the entire length of the channel is unaltered with the change, then there is definitely something wrong. If you're seeing just that the total temperature rise over the length of the channel is the same, then I'm guessing the channel temperature has approached the temperature of the HS by the time it exits. Which is it?

**Reply by Ethan-xj** [reply]
> then I'm guessing the channel temperature has approached the temperature of the HS by the time it exits

I output the steady result.
I check the result again. When I change the Phf, the temperature of flow channel is not changing. But the temperature of the HS is changing. So the heat flux changes indeed. But does this mean that Phf will not affect the amount of energy released into the flow channel? Or maybe I misunderstand sth?
The main question I care is I mentioned above:

> So if I want to build a model: a heat structure connected with two flow channels. How should I set the fraction of the power?

**Reply by joshuahansel** [reply]
> When I change the Phf, the temperature of flow channel is not changing. But the temperature of the HS is changing.

When you say the temperature of the flow channel is not changing, do you mean _absolutely_ not changing? Or is the temperature distribution just _approximately_ the same? The statement you've made implied that there is a lack of energy conservation, but that's only if there is absolutely no difference. If the channel is moving fast or has a lot of thermal mass, you made not see much difference.

> So if I want to build a model: a heat structure connected with two flow channels. How should I set the fraction of the power?

Your only option with this setup is to do as you have been: you set the different heated perimeters for each channel. I wouldn't expect the split to be exactly 1/3 and 2/3, but I think it should be roughly this.

> Phf*volumetric heat source should be the linear power density

This isn't correct. This would give you some number W/m^2.

**Reply by Ethan-xj** [reply]
It‘s approximately the same. OK. Maybe it‘s because the velocity I set is high.
And I forgot the HS length. It should times the length of the HS to get the linear power density.
I ask this question because the temperature of the flow channel is not correct in a 78-channels sodium came. And I will keep testing the question we discuss.
